githubmarkdownui/blocks/table.py

```diff
@@ -37,4 +37,42 @@ def table(content: List[List[str]], alignment: List[Optional[TableAlignment]] =
     :raises: TableAlignmentError when the alignment parameter is not the same length as the sublists of content
     :raises: TableDimensionError when the sublists of content do not contain the same number of elements
     """
-    pass
+    # Check if the inputs are valid. The content parameter must be a m x n matrix and alignment must be the same length as
+    # one element in content.
+    is_valid_table = all(isinstance(row, list) and len(row) == len(content[0]) for row in content)
+    if not is_valid_table:
+        raise TableDimensionError('Each row in the table must have the same number of columns')
+
+    if alignment and len(alignment) != len(content[0]):
+        raise TableAlignmentError('The alignment parameter is not the same length as a row in the table')
+
+    table_syntax = '<table><thead><tr>'
+
+    # Build the table header.
+    for i in range(len(content[0])):
+        if alignment and alignment[i]:
+            table_syntax += f'<th align="{alignment[i].value}">'
+        else:
+            table_syntax += '<th>'
+
+        table_syntax += f'{content[0][i]}</th>'
+
+    table_syntax += '</tr></thead><tbody>'
+
+    # Build the table body.
+    for i in range(1, len(content)):
+        table_syntax += '<tr>'
+
+        for j in range(len(content[i])):
```

Comment:
Same here

Reply:
Used enumerate and also realized the outer loop didn't need to track an index so it's more simple to read now